rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParser.java

```diff
@@ -246,6 +246,18 @@ public Builder classpathFromResource(ExecutionContext ctx, String... artifactNam
             return this;
         }
 
+        /**
+         * This is an internal API which is subject to removal or change.
+         */
+        public Builder addClasspathEntry(Path classpath) {
+            if (this.classpath.isEmpty()) {
+                this.classpath = Collections.singletonList(classpath);
+            } else {
+                this.classpath.add(classpath);
+            }
```

Comment:
`Collections.singletonList()` returns an immutable list. So if this were to be called twice it would fail. Perhaps use `ListUtils.concat()` in the `else` clause.